MAG does not start up if URL for xca-38 is not defined

if not provided in application.yaml MAG won't start.

org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'iti67RouteBuilder' defined in URL [jar:file:/home/mag/mag.jar!/BOOT-INF/classes!/ch/bfh/ti/i4mi/mag/mhd/iti67/Iti67RouteBuilder.class]: Failed to instantiate [ch.bfh.ti.i4mi.mag.mhd.iti67.Iti67RouteBuilder]: Constructor threw exception
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:321) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:309) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1395) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1232) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:569) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:529) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:373) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1222) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1188) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1123) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987) ~[spring-context-6.2.10.jar!/:6.2.10]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627) ~[spring-context-6.2.10.jar!/:6.2.10]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.5.5.jar!/:3.5.5]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752) ~[spring-boot-3.5.5.jar!/:3.5.5]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439) ~[spring-boot-3.5.5.jar!/:3.5.5]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:318) ~[spring-boot-3.5.5.jar!/:3.5.5]
        at ch.bfh.ti.i4mi.mag.MobileAccessGateway.main(MobileAccessGateway.java:54) ~[classes!/:na]
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[mag.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:108) ~[mag.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[mag.jar:na]
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88) ~[mag.jar:na]
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [ch.bfh.ti.i4mi.mag.mhd.iti67.Iti67RouteBuilder]: Constructor threw exception
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:223) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:146) ~[spring-beans-6.2.10.jar!/:6.2.10]
        at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:318) ~[spring-beans-6.2.10.jar!/:6.2.10]
        ... 25 common frames omitted
Caused by: java.lang.IllegalArgumentException: URI must not be null
        at org.springframework.util.Assert.notNull(Assert.java:181) ~[spring-core-6.2.10.jar!/:6.2.10]
        at org.springframework.web.util.UriComponentsBuilder.fromUriString(UriComponentsBuilder.java:182) ~[spring-web-6.2.10.jar!/:6.2.10]
        at ch.bfh.ti.i4mi.mag.common.MagRouteBuilder.buildOutgoingEndpoint(MagRouteBuilder.java:29) ~[classes!/:na]
        at ch.bfh.ti.i4mi.mag.mhd.iti67.Iti67RouteBuilder.<init>(Iti67RouteBuilder.java:70) ~[classes!/:na]
        at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(Unknown Source) ~[na:na]
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Unknown Source) ~[na:na]
        at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source) ~[na:na]
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:210) ~[spring-beans-6.2.10.jar!/:6.2.10]
        ... 27 common frames omitted


Suggest that if xca-38 is not defined that we don't query it.

--- a/docs/changelog.md
+++ b/docs/changelog.md
@@ -3,6 +3,7 @@
 ## v2.0.2 - unreleased
 
 - Enable https with client certificates and without ([#19](https://github.com/ahdis/MobileAccessGateway/issues/19)).
+- Fix startup issue when XCA endpoints are not defined ([#18](https://github.com/ahdis/MobileAccessGateway/issues/18)).
 
 ## v2.0.1 - 2025/09/30
 

--- a/src/main/java/ch/bfh/ti/i4mi/mag/mhd/iti67/Iti67RouteBuilder.java
+++ b/src/main/java/ch/bfh/ti/i4mi/mag/mhd/iti67/Iti67RouteBuilder.java
@@ -43,7 +43,7 @@ import static org.openehealth.ipf.platform.camel.ihe.fhir.core.FhirCamelTranslat
  * IHE MHD: Find Document References [ITI-67] for Document Responder https://oehf.github.io/ipf-docs/docs/ihe/iti67/
  */
 @Component
-@ConditionalOnProperty({"mag.xds.iti-18", "mag.xds.iti-57"})
+@ConditionalOnProperty({"mag.xds.iti-18", "mag.xds.xca-38", "mag.xds.iti-57"})
 class Iti67RouteBuilder extends MagRouteBuilder {
     private static final Logger log = org.slf4j.LoggerFactory.getLogger(Iti67RouteBuilder.class);
     private final MagXdsProps xdsProps;

--- a/src/main/java/ch/bfh/ti/i4mi/mag/mhd/iti68/Iti68RouteBuilder.java
+++ b/src/main/java/ch/bfh/ti/i4mi/mag/mhd/iti68/Iti68RouteBuilder.java
@@ -31,7 +31,7 @@ import org.springframework.stereotype.Component;
  * https://oehf.github.io/ipf-docs/docs/boot-fhir/ https://camel.apache.org/components/latest/servlet-component.html
  */
 @Component
-@ConditionalOnProperty("mag.xds.iti-43")
+@ConditionalOnProperty({"mag.xds.iti-43", "mag.xds.xca-39"})
 class Iti68RouteBuilder extends MagRouteBuilder {
     private static final Logger log = LoggerFactory.getLogger(Iti68RouteBuilder.class);
 

